extra/gemm/simple_matmul.py

```diff
@@ -2,14 +2,14 @@
 from tinygrad.helpers import getenv
 from tinygrad import dtypes, Tensor
 dtype_in = dtypes.half if getenv("HALF") else dtypes.float
+acc_dtype = dtypes.half if getenv("ACC_HALF") else None
 N = getenv("N", 4096)
 CNT = getenv("CNT", 10)
-a, b = Tensor.rand(N, N, dtype=dtype_in).realize(), Tensor.rand(N, N, dtype=dtype_in).realize()
+a, b = Tensor.rand(N, N, dtype=dtypes.float).realize().cast(dtype=dtype_in).realize(), Tensor.rand(N, N, dtype=dtypes.float).realize().cast(dtype=dtype_in).realize()
 for i in range(CNT):
   if i > 0 and getenv("RAND", 0) != 0:
-    a, b = Tensor.rand(N, N, dtype=dtype_in).realize(), Tensor.rand(N, N, dtype=dtype_in).realize()
-  # NOTE: accumulate is in float32
-  c = (a @ b).realize()
+    a, b = Tensor.rand(N, N, dtype=dtypes.float).realize().cast(dtype=dtype_in).realize(), Tensor.rand(N, N, dtype=dtypes.float).realize().cast(dtype=dtype_in).realize()
+  c = a.matmul(b, acc_dtype=acc_dtype).realize()
 comp = a.numpy().astype(np.float32) @ b.numpy().astype(np.float32)
```

Comment:
if i run `HALF=1 ACC_HALF=1 DEBUG=2 python3 extra/gemm/simple_matmul.py`, it failed due to outputs not matched

Reply:
If you use a smaller size like `N=1024`, it will be correct.  It's being compared to numpy float32 results, but the half precisions loss really creeps in above that.  Alternatively you can change the numpy comp to use half as well, but it gets very very slow (I never got a `N=4096`) to finish.